sourcefiles/randoconfig.py

```diff
@@ -1236,8 +1236,8 @@ def enum_enum_dict(d):
                 for boss_id in boss_ids
         }
 
-        boss_details_dict[str(BossID.MAGUS)]['character'] = self.magus_char
-        boss_details_dict[str(BossID.BLACK_TYRANO)]['element'] = self.black_tyrano_element
+        boss_details_dict[str(BossID.MAGUS)]['character'] = str(self.magus_char)
+        boss_details_dict[str(BossID.BLACK_TYRANO)]['element'] = str(self.black_tyrano_element)
```

Comment:
This was correct when you made the PR, but it's a little different now to avoid duplicate data in the config (because of your prior comments).

- To get Magus's assignment, just check `config.enemy_dict[EnemyID.Magus].name`
- To get the tyrano element, you can call `bossrando.get_black_tyrano_element(config)` as it is now in the human-readable spoilers.  This reaches into the tyrano's AI to get the element instead of having a duplicate standalone entry.